.github/workflows/request-reviews.yml

```diff
@@ -55,15 +55,38 @@ jobs:
           TARGET_BRANCH: ${{ github.event.pull_request.base.ref }}
           WORKSPACE_PATH: ${{ github.workspace }}
         run: |
+            import logging
             import os
             import sys
             sys.path.append(os.path.join(os.environ['WORKSPACE_PATH'], ".github"))
+            from edk2toollib.utility_functions import RunCmd
+            from io import StringIO
             from scripts import GitHub
 
             WORKSPACE_PATH = os.environ['WORKSPACE_PATH']
             GET_MAINTAINER_LOCAL_PATH = os.path.join(WORKSPACE_PATH, os.environ['GET_MAINTAINER_REL_PATH'])
 
-            reviewers = GitHub.get_reviewers_for_current_branch(WORKSPACE_PATH, GET_MAINTAINER_LOCAL_PATH, f"origin/{os.environ['TARGET_BRANCH']}")
+            pr_commit_sha = GitHub.get_pr_sha(os.environ['GH_TOKEN'], os.environ['ORG_NAME'], os.environ['REPO_NAME'], os.environ['PR_NUMBER'])
+            if not pr_commit_sha:
+                sys.exit(1)
+
+            print(f"::notice title=PR Commit SHA::Looking at files in consolidated PR commit: {pr_commit_sha}")
+
+            out_stream_buffer = StringIO()
+            cmd_ret = RunCmd(
+                "git",
+                f"fetch origin {pr_commit_sha}",
+                workingdir=WORKSPACE_PATH,
+                outstream=out_stream_buffer,
+                logging_level=logging.INFO,
+            )
+            if cmd_ret != 0:
+                print(
+                    f"::error title=Commit Fetch Error!::Error fetching PR commit: [{cmd_ret}]: {out_stream_buffer.getvalue()}"
+                )
+                sys.exit(1)
+
+            reviewers = GitHub.get_reviewers_for_range(WORKSPACE_PATH, GET_MAINTAINER_LOCAL_PATH, pr_commit_sha, pr_commit_sha)

```

Comment:
The start and end of the range is a single sha value, which means it will only collect reviewers from a single commit.  This doe snot look like it would collect all the reviewers for a series of commits in a PR.

Reply:
Because of the `pull_request_target` context, the SHA's in the workflow context change. For reference:

- [`pull_request` context](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows#pull_request)
- [`pull_request_target` context](https://docs.github.com/actions/using-workflows/events-that-trigger-workflows#pull_request_target)

For example, in the context of `pull_request` the `GITHUB_SHA` and `GITHUB_REF` refer to the PR merge commit/branch. In `pull_request_trigger`, they refer to the PR base commit/branch (e.g. `master`) since that is what is checked out in the context of the workflow.

In order for GitHub to check if a PR branch needs to be rebased or its state in general relative to the base branch, it creates a merge commit of the PR changes to the base. This happens on changes to the base branch (`master`) and on PR events like `synchronize` when pushes are made to the PR. Originally, I did use the range syntax to get a commit delta from the base origin point with `git log` but I found this commit would cover the same files and be more efficient especially in the context of information with the new trigger type.

I pushed several commits to a PR on my fork and verified they were reflected in the merge commit upon PR update.

---

This page has one of the best explanations I've found [The Many SHAs of a GitHub Pull Request](https://www.kenmuse.com/blog/the-many-shas-of-a-github-pull-request/).

Reply:
I did not realize it was the merge commit.  I agree this works.  I am now confused why other parts of this change include loops over the set of commits if the single merge commit is the only one really used.

Reply:
I left some of the logic to support that in place as an option in case this moves in a way that needs to be used in the future. I can drop anything not absolutely needed if you'd prefer that.

Reply:
@mdkinney, thanks for the feedback. Please let me know if you have a preference.

Reply:
I'd prefer to leave some of the generic functionality in place like the ability to take a commit range and iterate on it and clean up any sections of code not directly used in the solution when I transition to using PyGithub if that's okay with you.

Reply:
Please reopen if you have a strong opinion. Otherwise, I will go ahead and get this in and make any non-essential adjustments when I move the script to PyGithub.